src/aspire/utils/resolution_estimation.py

```diff
@@ -0,0 +1,394 @@
+        )
+
+    @property
+    def estimated_resolution(self):
+        """
+        Return estimated resolution of stacks `a` cross `b`.
+
+        :return: Numpy array.
+        """
+        self.analyze_correlations()
+        return self._resolutions
+
+    def analyze_correlations(self):
+        """
+        Convert from the Fourier Correlations to frequencies and resolution.
+        """
+        # `_analyzed` attribute in conjunction with `cutoff` allow a
+        # user to try different cutoffs without recomputing the
+        # correlations (FFT/NUFFT calls).
+        if self._analyzed:
+            return
+
+        c_inds = np.zeros(self.correlations.shape[:-1], dtype=int)
+
+        # All correlations are above cutoff,
+        #   set index of highest sampled frequency.
+        c_inds[np.min(self.correlations, axis=-1) > self.cutoff] = self.L // 2
+
+        # Correlations cross the cutoff.
+        # Find the first index of a correlation at `cutoff`.
+        # Should return 0 if not found, which corresponded to the case
+        # where all correlations are below cutoff.
+        c_ind = np.maximum(c_inds, np.argmax(self.correlations <= self.cutoff, axis=-1))
+
+        # Convert indices to frequency (as 1/Angstrom)
+        frequencies = self._freq(c_ind)
+
+        # Convert to resolution in Angstrom, smaller is higher frequency.
+        self._resolutions = 1 / frequencies
+
+    def _freq(self, k):
+        """
+        Converts `k` from index of Fourier transform to frequency (as
+        length 1/A).
+
+        :param k: Frequency index, integer or Numpy array of ints.
+        :return: Frequency in 1/Angstrom.
+        """
+
+        # From Shannon-Nyquist, for a given pixel-size, sampling theorem
+        # limits us to the sampled frequency 1/pixel_size.  Thus the
+        # Bandwidth ranges from `[-1/pixel_size, 1/pixel_size]`, so the
+        # total bandwidth is `2*(1/pixel_size)`.
+
+        # Given a real space signal observed with `L` bins
+        # (pixels/voxels), each with a `pixel_size` in Angstrom, we can
+        # compute the width of a Fourier space bin to be the `Bandwidth
+        # / L = (2*(1/pixel_size)) / L`.  Thus the frequency at an index
+        # `k` is `freq_k = k * 2 * (1 / pixel_size) / L = k * 2 /
+        # (pixel_size * L)
+
+        # _freq(k) Units: 1 / (pixels * (Angstrom / pixel) = 1 / Angstrom
+        # Similar idea to wavenumbers (cm-1).  Larger is higher frequency.
+        return k * 2 / (self.L * self.pixel_size)
+
+    def plot(self, save_to_file=False):
+        """
+        Generates a Fourier correlation plot.
+
+        :param save_to_file: Optionally, save plot to file.
+            Defaults False, enabled by providing a string filename.
+            User is responsible for providing reasonable filename.
+            See `https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.savefig.html`.
+        """
```

Comment:
I'll add it since you asked, representing a bulk of our users 😂; and I'm trying to be more positive here.  

While I ostensibly agree with what you are saying, the generalized FSC is not functionally complete without another (med-large) component, and so I generally would have optimized aesthetics and syntactic sugar later, after feature completeness, given no other feedback.